plonky2/src/plonk/prover.rs

```diff
@@ -58,13 +60,20 @@ pub fn set_lookup_wires<
 
         // Compute multiplicities.
         let mut multiplicities = vec![0; lut_len];
+
+        let mut table_value_to_idx: HashMap<u16, usize> = HashMap::new();
+
+        for (i, (inp_target, _)) in common_data.luts[lut_index].iter().enumerate() {
```

Comment:
I think this can be done more idiomatically with `.map()` and `.collect()

Reply:
How does [this](https://github.com/onsen-egg/plonky2/blob/a67cfdcb676b8637110621a8fe35251f0c583ec7/plonky2/src/plonk/prover.rs#L64-L68) look instead?